src/com/google/enterprise/adaptor/database/UniqueKey.java

```diff
@@ -75,22 +77,22 @@
       log.fine("element: `" + e + "'");
       e = e.trim();
       String def[] = e.split(":", 2);
-      if (2 != def.length) {
-        String errmsg = "Invalid UniqueKey definition for '" + e + "'. Valid"
-            + " definition is 'column_name:type' where types can be int, string,"
-            + " timestamp, date, time and long.";
-        throw new InvalidConfigurationException(errmsg);
-      }
-      def[0] = def[0].trim();
-      def[1] = def[1].trim();
-      String name = def[0];
+      String name = def[0].trim();
       ColumnType type;
-      try {
-        type = ColumnType.valueOf(def[1].toUpperCase(US));
-      } catch (IllegalArgumentException iae) {
-        String errmsg = "Invalid UniqueKey type '" + def[1] + "'. Valid"
-            + " types are: " + ColumnType.values();
-        throw new InvalidConfigurationException(errmsg);
+
+      if (def.length == 1) {
+        // No type given. Try to get type from ResultSetMetaData later.
+        type = ColumnType.UNKNOWN;
+      } else {
+        String typeStr = def[1].trim();
+        try {
+          type = ColumnType.valueOf(typeStr.toUpperCase(US));
+        } catch (IllegalArgumentException iae) {
+          String errmsg = "Invalid UniqueKey type '" + typeStr + "' for "
+              + "name. Valid types are: int, string, timestamp, date, time, "
```

Comment:
I had thought to comment previously on the uppercase nature of ColumnType.values(), but didn't. I thought it was better than listing the names like this, and not worth lowercasing the values for the log message.